qemu/tests/virtio_serial_throttling.py

```diff
@@ -0,0 +1,54 @@
+from avocado.utils import process
+from virttest import error_context
+
+from qemu.tests.virtio_serial_file_transfer import transfer_data
+
+import socket
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Induce throttling by opening chardev but not reading data from it:
+    1) Start guest with virtio-serial with unix option
+    2) On the host, open the chardev but don't read from it
+    3) In the guest, write to the virtio-serial port.
+    4) repeat step 1 to 3 with tcp option.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_chardev_params(vm, serial_id):
+        serial_device = vm.devices.get(serial_id)
+        chardev_id = serial_device.get_param('chardev')
+        chardev_device = vm.devices.get(chardev_id)
+        chardev_params = chardev_device.params
+        return chardev_params
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    path = '/tmp/port0'
+    sock = socket.socket(socket.AF_UNIX)
+    sock.connect(path)
+    process.run('touch /tmp/port0')
```

Comment:
First, do not need to define the `chardev_path` in cfg
Second, do not need to touch the file 'port0', vm can create the file automatically as long as the path `/tmp` exist.

And, it is not a good style, define params in cfg and script two times with the same value,  `params.get` instead.